frontend/app/.server/domain/services/application-year.service.ts

```diff
@@ -52,9 +53,29 @@ export class DefaultApplicationYearService implements ApplicationYearService {
   }
 
   private init(): void {
+    this.log.debug('Cache TTL value: %d ms', 1000 * this.serverConfig.LOOKUP_SVC_APPLICATION_YEAR_CACHE_TTL_SECONDS);
+
+    this.findRenewalApplicationYear = moize(this.findRenewalApplicationYear, {
+      matchesKey: (cacheKey, key) => cacheKey.every((item, idx) => item.date === key[idx].date),
```

Comment:
What the following line do?

Reply:
It caches according to the current date, ignoring the passed-in user id..

I left a `TODO` suggesting we remove the user id since it isn't required, but that change would be too big for this PR.